integration-tests/src/test/java/uk/gov/di/authentication/api/OrchestrationToAuthenticationAuthorizeIntegrationTest.java

```diff
@@ -0,0 +1,385 @@
+                txmaAuditQueue, List.of(AUTHORISATION_REQUEST_RECEIVED, AUTHORISATION_INITIATED));
+    }
+
+    @Test
+    void
+            shouldSendSecureJarToAuthenticationWithRelevantScopesAndAddAccountManagementClaimWhenAmScopeIsPresent()
+                    throws ParseException, JOSEException, java.text.ParseException {
+        var rpRequestedScopes = new Scope(OPENID, PHONE, EMAIL, ACCOUNT_MANAGEMENT);
+        registerClient(rpRequestedScopes.toStringList(), false);
+
+        var response =
+                makeRequest(
+                        Optional.empty(),
+                        constructHeaders(Optional.empty()),
+                        constructQueryStringParameters(rpRequestedScopes.toString(), null));
+
+        var authorizationRequest =
+                validateQueryRequestToAuthenticationAndReturnAuthRequest(response);
+        var encryptedRequestObject = authorizationRequest.getRequestObject();
+        var signedJWTResponse = decryptJWT((EncryptedJWT) encryptedRequestObject);
+        validateStandardClaimsInJar(signedJWTResponse);
+
+        assertThat(
+                Objects.nonNull(signedJWTResponse.getJWTClaimsSet().getClaim("claim")),
+                equalTo(true));
+        var claimsRequest =
+                OIDCClaimsRequest.parse(
+                        (String) signedJWTResponse.getJWTClaimsSet().getClaim("claim"));
+        assertThat(
+                Objects.nonNull(claimsRequest.getUserInfoClaimsRequest().get("public_subject_id")),
+                equalTo(true));
+        assertTxmaAuditEventsReceived(
+                txmaAuditQueue, List.of(AUTHORISATION_REQUEST_RECEIVED, AUTHORISATION_INITIATED));
+    }
+
+    @Test
+    void
+            shouldSendSecureJarToAuthenticationWithRelevantScopesAndAddGovUkAccountClaimWhenGovUkScopeIsPresent()
+                    throws ParseException, JOSEException, java.text.ParseException {
+        var rpRequestedScopes = new Scope(OPENID, PHONE, EMAIL, GOVUK_ACCOUNT);
+        registerClient(rpRequestedScopes.toStringList(), false);
+
+        var response =
+                makeRequest(
+                        Optional.empty(),
+                        constructHeaders(Optional.empty()),
+                        constructQueryStringParameters(rpRequestedScopes.toString(), null));
+
+        var authorizationRequest =
+                validateQueryRequestToAuthenticationAndReturnAuthRequest(response);
+        var encryptedRequestObject = authorizationRequest.getRequestObject();
+        var signedJWTResponse = decryptJWT((EncryptedJWT) encryptedRequestObject);
+
+        validateStandardClaimsInJar(signedJWTResponse);
+        assertThat(
+                Objects.nonNull(signedJWTResponse.getJWTClaimsSet().getClaim("claim")),
+                equalTo(true));
+        var claimsRequest =
+                OIDCClaimsRequest.parse(
+                        (String) signedJWTResponse.getJWTClaimsSet().getClaim("claim"));
+        assertThat(
```

Comment:
Obviously this is a bit of a new domain for me, but I'm not entirely clear from reading some of these tests what is different in each of the assertions, and what the important thing that's changing is. For example, in this case the test is about the GOVUK_ACCOUNT, but without context it's not clear to me which of these assertions is about that (is it the legacy_subject_id?) and therefore what's the important bit of this test.

I wonder if it could be made a bit clearer by a) abstracting away a bit of the noise in the tests, and b) if I'm correct about this being the important line, maybe extracting the legacy_subject_id as a variable (`govukAccountClaim`?)

Reply:
@BeckaL that's fair. I've pushed up a change to make it clearer